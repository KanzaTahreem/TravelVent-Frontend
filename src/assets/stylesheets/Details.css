.details {
  width: 100%;
  height: auto;
  justify-content: space-between;
  align-items: flex-start;
}

.bg_image {
  width: 60%;
  height: 100vh;
  position: relative;
  align-self: center;
}

.details-img {
  width: 100%;
  height: 99.8%;
  object-fit: cover;
}

.back_button {
  bottom: 50px;
  left: 0;
  border: none;
  padding: 19px 25px;
  color: var(--white);
  font-size: 25px;
  position: absolute;
  padding-bottom: 9px;
  border-top-right-radius: 50%;
  border-bottom-right-radius: 50%;
  background: var(--dark-lemon-lime);
}

.info {
  width: 38%;
  height: 100vh;
  align-items: flex-end;
  padding: 7rem 3rem 0 0;
}

.info-title {
  font-size: 3rem;
  margin-bottom: 1.5rem;
  text-transform: uppercase;
  color: var(--sea-blue);
  line-height: 34px;
  text-align: center;
}

.info-subtitle {
  margin-bottom: 30px;
}

.quote {
  padding-bottom: 2rem;
  font-style: italic;
  font-size: 1.2rem;
  color: var(--dark-lemon-lime);
}

.box {
  justify-content: space-between;
  align-items: flex-start;
  width: 100%;
  padding: 11px 15px;
  background-color: var(--albaster);
  font-size: 1.5rem;
}

.darkbox {
  justify-content: space-between;
  align-items: flex-start;
  width: 100%;
  padding: 11px 15px;
  font-size: 1.5rem;
  background-color: #dddbd8;
}

.description {
  width: 100%;
  padding: 20px 12px;
  font-size: 1.5rem;
  font-style: italic;
  text-transform: capitalize;
  color: var(--cadet-gray);
}

.home-reserve-btn {
  margin-top: 50px;
  padding: 15px;
  width: 160px;
  border: none;
  color: white;
  font-size: 18px;
  border-radius: 50px;
  align-items: center;
  justify-content: space-between;
  background: var(--dark-lemon-lime);
}

@media (max-width: 768px) {
  .details.row {
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
  }

  .bg_image {
    width: 100%;
    height: 30vh;
    margin-bottom: 20px;
  }

  .back_button {
    top: 10px;
    bottom: auto;
    right: 0;
    left: auto;
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
    border-bottom-left-radius: 50%;
    border-top-left-radius: 50%;
    position: fixed;
    z-index: 10;
  }

  .info {
    width: 90%;
    padding: 0 10px;
    justify-content: center;
    align-items: center;
  }

  .home-reserve-btn {
    margin-bottom: 100px;
  }

  .description {
    font-size: 14px;
    text-align: center;
  }
}

@media screen and (min-width: 768px) {
  .info-title {
    text-align: right;
  }

  .details {
    margin-left: 220px;
    width: auto;
  }
}

@media screen and (min-width: 1110px) {
  .info {
    width: 30%;
    padding: 7rem 4rem 0 0;
  }
}
