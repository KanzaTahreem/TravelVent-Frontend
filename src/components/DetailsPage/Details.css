.details {
  width: 100%;
  height: 100vh;
  justify-content: space-between;
  align-items: flex-start;
}

.bg_image {
  width: 50%;
  position: relative;
}

img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.back_button {
  position: absolute;
  bottom: 50px;
  left: 0;
  background-color: cornflowerblue;
  border: none;
  border-top-right-radius: 50%;
  border-bottom-right-radius: 50%;
  padding: 25px 30px;
  color: white;
}

.info {
  padding-right: 25px;
  width: 40%;
  justify-content: flex-end;
  align-items: flex-end;
}

.info-title {
  margin-bottom: 8px;
}

.info-subtitle {
  margin-bottom: 30px;
}

.box {
  justify-content: space-between;
  align-items: flex-start;
  width: 100%;
  padding: 20px 12px;
  background-color: #e5e5e5;
}

.darkbox {
  justify-content: space-between;
  align-items: flex-start;
  width: 100%;
  padding: 20px 12px;
  background-color: #a8a8a8;
}

.description {
  width: 100%;
  padding: 20px 12px;
  font-size: 18px;
}

.reserve-btn {
  margin-top: 50px;
  padding: 20px;
  width: 160px;
  border: none;
  background-color: cornflowerblue;
  color: white;
  font-size: 18px;
  border-radius: 50px;
  justify-content: space-between;
}
