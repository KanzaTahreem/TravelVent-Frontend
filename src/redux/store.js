import { configureStore } from '@reduxjs/toolkit';
import cartItems from './cartSlice';
import registrationSlice from './reducers/registrationSlice';

const store = configureStore({
  reducer: {
    cartItems,
    registration: registrationSlice,
  },
});

export default store;
