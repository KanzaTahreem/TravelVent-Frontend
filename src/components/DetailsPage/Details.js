// import { useParams } from 'react-router-dom';
// import { useSelector } from 'react-redux';
import { AiOutlineSetting } from 'react-icons/ai';
import { BsArrowRightCircle } from 'react-icons/bs';
import { IoChevronBackCircleOutline } from 'react-icons/io5';
import './Details.css';

// the commented code is for future use

export default function Details() {
  // const { id } = useParams();

  // const trips = useSelector((state) => state.trips.trips);

  // const trip = trips.find((trip) => trip.id === id);

  const reserveHandler = () => {
    // route to reserve page
  };

  return (
    <div className="details row">
      <div className="bg_image">
        <img
          src="https://images.pexels.com/photos/235985/pexels-photo-235985.jpeg?cs=srgb&dl=pexels-pixabay-235985.jpg&fm=jpg"
          alt="details"
          className="img"
        />
        <button className="back_button" type="button">
          <IoChevronBackCircleOutline />
        </button>
      </div>

      <div className="info column">
        <h1 className="info-title">Details</h1>
        <p className="info-subtitle">subtext</p>
        <div className="box row">
          <p>
            price
          </p>
          <p>
            100$
          </p>
        </div>

        <div className="darkbox row">
          <p>
            price
          </p>
          <p>
            100$
          </p>
        </div>

        <p className="description">
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam
        </p>

        <button className="reserve-btn row" type="button" onClick={reserveHandler}>
          <AiOutlineSetting />
          Reserve
          <BsArrowRightCircle />
        </button>
      </div>
    </div>
  );
}
