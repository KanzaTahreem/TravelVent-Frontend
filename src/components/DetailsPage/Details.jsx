import { useParams, Link } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { AiOutlineSetting } from 'react-icons/ai';
import { BsArrowRightCircle } from 'react-icons/bs';
import { IoChevronBackCircleOutline } from 'react-icons/io5';
import { fetchTrips } from '../../redux/tripSlice';
import '../../assets/stylesheets/Details.css';

// the commented code is for future use

export default function Details() {
  const { id } = useParams();

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchTrips());
  }, [dispatch]);

  const status = useSelector((state) => state.trips.status);
  const trips = useSelector((state) => state.trips.trips);

  if (status === 'loading') {
    return <div>Loading...</div>;
  }

  const trip = trips.find((trip) => trip.id === +id);

  let img = '';

  if (trip !== undefined && trip?.image_url !== undefined) {
    img = trip?.image_url;
  } else {
    img = 'https://rare-gallery.com/uploads/posts/558972-sky-high-resolution.jpg';
  }

  return (
    <div className="details row">
      <div className="bg_image">
        <img
          src={`${img}`}
          alt="details"
          className="img"
        />
        <Link className="back_button" to="/home">
          <IoChevronBackCircleOutline />
        </Link>
      </div>

      <div className="info column">
        <h1 className="info-title">{trip?.destination_city}</h1>
        <div className="box row">
          <p>
            Price
          </p>
          <p>
            {`${trip?.price} $`}
          </p>
        </div>

        <div className="darkbox row">
          <p>
            Ratings
          </p>
          <p>
            {`${trip?.rating}/5`}
          </p>
        </div>

        <p className="description">
          {trip?.description}
        </p>

        <Link className="home-reserve-btn row" to={`/home/${trip?.id}/reserve`}>
          <AiOutlineSetting />
          Reserve
          <BsArrowRightCircle />
        </Link>
      </div>
    </div>
  );
}
