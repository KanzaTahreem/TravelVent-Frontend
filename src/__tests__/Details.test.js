import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Details from '../components/Details';

describe('Testing Details', () => {
  let details;
  beforeAll(async () => {
    details = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Details />
          </Router>
        </Provider>,
      )
      .toJSON();
  });

  it('should renders correctly', () => {
    expect(details).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(details).toMatchSnapshot();
  });

  it('Should have a link to the reservation page', () => {
    const link = screen.findByTestId('reservation-link');
    waitFor(() => expect(link).toBeInTheDocument());
    waitFor(() => expect(link).toHaveAttribute('href'));
  });

  const mockTrips = [
    {
      id: 1,
      destination_city: 'New York',
      price: 500,
      rating: 4.5,
      description: 'Explore the Big Apple and all its wonders!',
      image_url: 'https://example.com/image.jpg',
    },
  ];

  jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useParams: () => ({ id: '1' }),
  }));

  it('Details of a trip are correctly displayed', () => {
    jest.spyOn(store, 'getState').mockReturnValue({
      trips: {
        status: 'succeeded',
        trips: mockTrips,
      },
    });

    waitFor(() => expect(getByAltText('details')).toHaveAttribute('src', 'https://example.com/image.jpg'));
    waitFor(() => expect(getByText('New York')).toBeInTheDocument());
    waitFor(() => expect(getByText('Price')).toBeInTheDocument());
    waitFor(() => expect(getByText('$500')).toBeInTheDocument());
    waitFor(() => expect(getByText('Ratings')).toBeInTheDocument());
    waitFor(() => expect(getByText('4.5/5')).toBeInTheDocument());
    waitFor(() => expect(getByText('Explore the Big Apple and all its wonders!')).toBeInTheDocument());
    waitFor(() => expect(getByText('Reserve')).toBeInTheDocument());
  });
});
