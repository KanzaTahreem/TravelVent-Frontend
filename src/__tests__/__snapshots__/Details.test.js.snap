// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Details should match the snapshot 1`] = `
<div
  className="details row"
>
  <div
    className="bg_image"
  >
    <img
      alt="details"
      className="details-img"
      src="https://rare-gallery.com/uploads/posts/558972-sky-high-resolution.jpg"
    />
    <a
      className="back_button"
      href="/home"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 64C150 64 64 150 64 256s86 192 192 192 192-86 192-192S362 64 256 64z"
          fill="none"
          strokeMiterlimit="10"
          strokeWidth="32"
        />
        <path
          d="M296 352l-96-96 96-96"
          fill="none"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="32"
        />
      </svg>
    </a>
  </div>
  <div
    className="info column"
  >
    <h1
      className="info-title"
    />
    <p
      className="quote"
    >
      Better to see something once than to hear about it a thousand times.
    </p>
    <div
      className="box row"
    >
      <p>
        Price
      </p>
      <p>
        undefined $
      </p>
    </div>
    <div
      className="darkbox row"
    >
      <p>
        Ratings
      </p>
      <p>
        undefined/5
      </p>
    </div>
    <p
      className="description"
    />
    <a
      className="home-reserve-btn row"
      data-testid="reservation-link"
      href="/reserve/undefined"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M924.8 625.7l-65.5-56c3.1-19 4.7-38.4 4.7-57.8s-1.6-38.8-4.7-57.8l65.5-56a32.03 32.03 0 0 0 9.3-35.2l-.9-2.6a443.74 443.74 0 0 0-79.7-137.9l-1.8-2.1a32.12 32.12 0 0 0-35.1-9.5l-81.3 28.9c-30-24.6-63.5-44-99.7-57.6l-15.7-85a32.05 32.05 0 0 0-25.8-25.7l-2.7-.5c-52.1-9.4-106.9-9.4-159 0l-2.7.5a32.05 32.05 0 0 0-25.8 25.7l-15.8 85.4a351.86 351.86 0 0 0-99 57.4l-81.9-29.1a32 32 0 0 0-35.1 9.5l-1.8 2.1a446.02 446.02 0 0 0-79.7 137.9l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.3 56.6c-3.1 18.8-4.6 38-4.6 57.1 0 19.2 1.5 38.4 4.6 57.1L99 625.5a32.03 32.03 0 0 0-9.3 35.2l.9 2.6c18.1 50.4 44.9 96.9 79.7 137.9l1.8 2.1a32.12 32.12 0 0 0 35.1 9.5l81.9-29.1c29.8 24.5 63.1 43.9 99 57.4l15.8 85.4a32.05 32.05 0 0 0 25.8 25.7l2.7.5a449.4 449.4 0 0 0 159 0l2.7-.5a32.05 32.05 0 0 0 25.8-25.7l15.7-85a350 350 0 0 0 99.7-57.6l81.3 28.9a32 32 0 0 0 35.1-9.5l1.8-2.1c34.8-41.1 61.6-87.5 79.7-137.9l.9-2.6c4.5-12.3.8-26.3-9.3-35zM788.3 465.9c2.5 15.1 3.8 30.6 3.8 46.1s-1.3 31-3.8 46.1l-6.6 40.1 74.7 63.9a370.03 370.03 0 0 1-42.6 73.6L721 702.8l-31.4 25.8c-23.9 19.6-50.5 35-79.3 45.8l-38.1 14.3-17.9 97a377.5 377.5 0 0 1-85 0l-17.9-97.2-37.8-14.5c-28.5-10.8-55-26.2-78.7-45.7l-31.4-25.9-93.4 33.2c-17-22.9-31.2-47.6-42.6-73.6l75.5-64.5-6.5-40c-2.4-14.9-3.7-30.3-3.7-45.5 0-15.3 1.2-30.6 3.7-45.5l6.5-40-75.5-64.5c11.3-26.1 25.6-50.7 42.6-73.6l93.4 33.2 31.4-25.9c23.7-19.5 50.2-34.9 78.7-45.7l37.9-14.3 17.9-97.2c28.1-3.2 56.8-3.2 85 0l17.9 97 38.1 14.3c28.7 10.8 55.4 26.2 79.3 45.8l31.4 25.8 92.8-32.9c17 22.9 31.2 47.6 42.6 73.6L781.8 426l6.5 39.9zM512 326c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 0 1 512 614c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 0 1 400 502c0-29.9 11.7-58 32.8-79.2C454 401.6 482.1 390 512 390c29.9 0 58 11.6 79.2 32.8A111.6 111.6 0 0 1 624 502c0 29.9-11.7 58-32.8 79.2z"
        />
      </svg>
      Reserve
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </a>
  </div>
</div>
`;
